specifyweb/specify/tree_views.py

```diff
@@ -359,6 +360,65 @@ def tree_rank_item_count(request, tree, rankid):
     count = tree_rank_count(tree, rank.id)
     return HttpResponse(toJson(count), content_type='application/json')
 
+TAXON_TREES = ["Mineral", "Rock", "Meteorite", "Fossil"]
+TAXON_RANKS = ["One", "Two", "Three"]
+
+# TODO: Add openapi schema for this endpoint
+@tree_mutation
+def add_geo_default_trees(request):
+    for tree in TAXON_TREES:
+        if Taxontreedef.objects.filter(name=tree).exists():
+            continue
+        tree_name = f"{tree} Taxon"
+        ttd = Taxontreedef.objects.create(name=tree_name)
+        ttd.save()
+        rank_id = 0
+        ttdi = None
+        for rank in TAXON_RANKS:
+            name = f"{tree} {rank}"
+            rank_id += 10
+            ttdi = Taxontreedefitem.objects.create(
+                name=name,
+                title=name,
+                rankid=rank_id,
+                parent=ttdi,
+                treedef=ttd,
+            )
+            ttdi.save()
+
+# TODO: Add openapi schema for this endpoint
+@tree_mutation
+def remove_geo_default_trees(request):
+    for tree in reversed(TAXON_TREES):
+        tree_name = f"{tree} Taxon"
+        ttd = Taxontreedef.objects.filter(name=tree_name)
+        if ttd.exists():
+            for rank in reversed(TAXON_RANKS):
+                name = f"{tree} {rank}"
+                ttdi = Taxontreedefitem.objects.filter(
+                    name=name, treedef=ttd.first()
+                )
+                if ttdi.exists():
+                    try:
+                        ttdi.delete()
+                    except Exception as e:
+                        print(f"Error deleting taxontreedefitem {name}: {e}")
+                        continue
+            
+            try:
+                root_item = Taxontreedefitem.objects.get(name=f"{tree} Root")
+                root_item.delete(allow_root_del=True)
+            except Taxontreedefitem.DoesNotExist:
+                print(f"Taxontreedefitem with name {tree} Root does not exist.")
+            except Exception as e:
+                print(f"Error deleting taxontreedefitem {tree} Root: {e}")
```

Comment:
improve this. don't do an exception catch and print it out. propagate it up. I'm not sure what the expected behaviour really is here too. For starters, you don't need to make a distinction for your taxon trees. Make it so that frontend can give you a treedefid, and the tree type (taxon or geography or whatever), and you delete all tree ranks from it. 

Better yet, remove the treedef blocker from treedefinition and then do it, your life will be simpler, plus, this code will be actually useful rather than a minor edge case (any body will be able to delele treedef)